kyuubi-server/web-ui/src/views/editor/components/Editor.vue

```diff
@@ -0,0 +1,261 @@
+* with the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+-->
+
+<template>
+  <div class="editor">
+    <el-space>
+      <el-select v-model="param.engineType" :placeholder="$t('engine_type')">
+        <el-option
+          v-for="item in getEngineType()"
+          :key="item"
+          :label="item"
+          :value="item" />
+      </el-select>
+      <el-button
+        :disabled="!param.engineType || !editorVariables.content"
+        :loading="resultLoading"
+        type="primary"
+        icon="VideoPlay"
+        @click="handleQuerySql">
+        {{ $t('operation.run') }}
+      </el-button>
+      <el-dropdown @command="handleChangeLimit">
+        <span class="el-dropdown-link">
+          Limit: {{ limit }}
+          <el-icon class="el-icon--right">
+            <arrow-down />
+          </el-icon>
+        </span>
+        <template #dropdown>
+          <el-dropdown-menu>
+            <el-dropdown-item v-for="l in limits" :key="l" :command="l">
+              Limit: {{ l }}
+            </el-dropdown-item>
+          </el-dropdown-menu>
+        </template>
+      </el-dropdown>
+    </el-space>
+    <section>
+      <MonacoEditor
+        v-model="editorVariables.content"
+        :language="editorVariables.language"
+        :theme="theme"
+        @editor-mounted="editorMounted"
+        @change="handleContentChange"
+        @editor-save="editorSave" />
+    </section>
+    <pre v-show="sqlLog" v-loading="logLoading">{{ sqlLog }}</pre>
+    <el-tabs v-model="activeTab" type="card" class="result-el-tabs">
+      <el-tab-pane
+        v-loading="resultLoading"
+        :label="`Result${sqlResult?.length ? ` (${sqlResult?.length})` : ''}`"
+        name="result">
+        <Result :data="sqlResult" />
+      </el-tab-pane>
+    </el-tabs>
+  </div>
+</template>
+
+<script lang="ts" setup>
+  import MonacoEditor from '@/components/monaco-editor/index.vue'
+  import Result from './Result.vue'
+  import { ref, reactive, onUnmounted, toRaw } from 'vue'
+  import type { Ref } from 'vue'
+  import * as monaco from 'monaco-editor'
+  import { format } from 'sql-formatter'
+  import { ElMessage } from 'element-plus'
+  import { useI18n } from 'vue-i18n'
+  import { getEngineType } from '@/utils/engine'
+  import {
+    openSession,
+    closeSession,
+    runSql,
+    getSqlRowset,
+    getSqlMetadata,
+    log
+  } from '@/api/editor'
+  import type { IResponse, ISqlResult, IFields, ILog } from './types'
+
+  const { t } = useI18n()
+  const param = reactive({
+    engineType: 'SPARK_SQL'
+  })
+  const limit = ref(10)
+  const limits = ref([10, 50, 100])
```

Comment:
We don't need to bind this limits two-way data binding, simple constants should suffice.